plugins/modules/cloud/misc/terraform.py

```diff
@@ -147,30 +154,55 @@
   returned: always
   sample: terraform apply ...
 """
-
 import os
 import json
 import tempfile
-import traceback
-from ansible.module_utils.six.moves import shlex_quote
+import csv
 
+from datetime import datetime
+from io import StringIO
+from ansible.module_utils.six.moves import shlex_quote
 from ansible.module_utils.basic import AnsibleModule
 
 DESTROY_ARGS = ('destroy', '-no-color', '-force')
 APPLY_ARGS = ('apply', '-no-color', '-input=false', '-auto-approve=true')
 module = None
+current_DateTime = datetime.now().strftime("%d/%m/%Y %H:%M:%S")
 
 
-def preflight_validation(bin_path, project_path, variables_args=None, plan_file=None):
+def write_error_to_file(log_error_path, content):
```

Comment:
hi @FrancisBilla I am not sure that writing a log file (on the target machine) is such a good idea. I believe a better mechanism is to return the errors in the module's output.